### ISSUE TITLE
Segfault when figures are deleted in random order

### ISSUE THREAD

**Issue by olcc** [matplotlib/matplotlib#6231]
Hello, please consider the following code...
```
import matplotlib.pyplot as pyplot
fig1 = pyplot.figure()
fig2 = pyplot.figure()
pyplot.show()
```
If I delete Figure 2, then Figure 1, the program finishes correctly.
If I delete Figure 1, then Figure 2, the program finishes with a segmentation error.
I believe that the program should not segfault in the second case.

Olivier
-- Matplotlib 1.5.1, Python 3.5.1, Debian Linux

**Comment by tacaswell** [comment]
What backend are you using?

**Comment by olcc** [comment]
Using matplotlib backend: TkAgg

**Comment by tacaswell** [comment]
I can not reproduce this with 1.5.1 / 3.5.1 from anaconda on arch.

**Comment by cgohlke** [comment]
This is a known issue on Python 3.4 for Windows, which ships with an outdated version of tcl86. Maybe that's the case here too? See https://github.com/ipython/ipython/issues/8921 and http://stackoverflow.com/questions/30729180

**Comment by olcc** [comment]
Yes, it's the same behavior, but on Linux. I have these packages :
* tcl     8.6.0+9
* tk      8.6.0+9

**Comment by cgohlke** [comment]
On Windows tcl/tk 8.6.1 (included with Python 3.4) crashes while tcl/tk 8.6.4 (included with Python 3.5) does not crash.

**Comment by tacaswell** [comment]
I was testing with tcl/tk 8.5.0 (what anaconda ships) or 8.6.4 (what arch packages).

@sandrotosi This is something that may need to be addressed in the debian packaging?

**Comment by tacaswell** [comment]
Possibly relevant bug report

 - https://bugs.python.org/issue24858 (possibly other packaging issues with debian)

**Comment by tacaswell** [comment]
loosely related (tk segfaults due to bad linking) https://github.com/matplotlib/matplotlib/issues/3821

**Comment by tacaswell** [comment]
Can someone here who can reproduce this test #6277 ?

**Comment by sandrotosi** [comment]
@tacaswell sorry for getting back so late; I just tested on debian with tcl/tk (with TKAgg backend) and I cant reproduce this issue

**Comment by tacaswell** [comment]
What is the exact tk/tcl version you used?

**Comment by sandrotosi** [comment]
gaah sorry i thought i wrote that: 8.6.5 for tcl and tk

**Comment by tacaswell** [comment]
what debian channel are those versions available in?

**Comment by sandrotosi** [comment]
debian unstable (but the same version is available also in testing; stable has 8.6.2)

**Comment by mdboom** [comment]
Closing as fixed upstream.  We've documented the problematic versions, and I'm not sure if there's much we can do as a workaround.
